### PR TITLE
Add HTTP Strict-Transport-Security header to unknown URLs

### PR DESCRIPTION
**Pull request by MahmudH**
- This sets the HSTS header to the domain within VCL
- Enables consistency throughout the CDN.

Trello: https://trello.com/c/0TgCOgbX/3397-3-configure-fastly-to-always-append-an-hsts-header

### CODE REVIEW

File: modules/www/www.vcl.tftpl [hunk 1]
```diff
@@ -452,6 +452,11 @@ sub vcl_miss {
 }
 
 sub vcl_deliver {
+  # Add the HSTS header with a max-age of 5 mins to URLs that don't exist
+  if (!(resp.http.Strict-Transport-Security)) {
```
Comment (sengi) [inline modules/www/www.vcl.tftpl:456]: Why the conditional?

File: modules/www/www.vcl.tftpl [hunk 2]
```diff
@@ -452,6 +452,11 @@ sub vcl_miss {
...
+    set resp.http.Strict-Transport-Security = "max-age=300"; includeSubDomains; preload
```
Comment (sengi) [inline modules/www/www.vcl.tftpl:457]: HSTS with max-age=300 doesn't seem useful to me — what's the rationale here?

**Review by sengi — APPROVED** [review]
The idea SGTM, but I have a couple of practical concerns about merging this as-is:

- do we plan to remove the HSTS stuff from apps? (I think we'll need to, otherwise we're duplicating the config for this important header, which just seems unnecessarily risky.)
- why the super short max-age?

File: modules/www/www.vcl.tftpl (hunk 2 above)
Comment (sengi) [reply modules/www/www.vcl.tftpl:457]: I see from the notes on the internal-only Trello card that you plan to set this to 300 and then increase it later, but that doesn't make any sense to me given that we're already serving HSTS headers with an appropriately-long max-age (`strict-transport-security: max-age=31536000; preload`).

File: modules/www/www.vcl.tftpl (hunk 1 above)
Comment (MahmudH) [reply modules/www/www.vcl.tftpl:456]: The conditional is there to stop overriding the header so we would set it in the VCL when it's not set by the apps. 

E.g. https://www.gov.uk/talu-psatalu-psa does not have it set so this would cover these use cases.

**Comment by sengi** [comment]
tl;dr:
- HSTS affects a whole domain, not individual pages/paths.
- We currently have a slight inconsistency in how we're serving the HSTS header, for example we're not serving it on some 404 responses.
- We don't want to be serving conflicting or inconsistent HSTS headers.
- We should therefore configure the HSTS header in the CDN config so that it's trivial to see that it's correct.
- The value needs to be the same as the one we're currently serving, so we don't break stuff or make the inconsistency worse.

File: modules/www/www.vcl.tftpl [hunk 3]
```diff
@@ -452,6 +452,8 @@ sub vcl_miss {
 }
 
 sub vcl_deliver {
+  set resp.http.Strict-Transport-Security = "max-age=31536000"; preload
```
Comment (sengi) [inline modules/www/www.vcl.tftpl:455]: Have you tested this? It doesn't look like valid VCL syntax to me.

Comment (MahmudH) [reply modules/www/www.vcl.tftpl]: Just confirmed the syntax through:
https://fiddle.fastly.dev/fiddle/5c52bc2d

it should be:
set resp.http.Strict-Transport-Security = "max-age=31536000; preload";

**Review by sengi — APPROVED** [review]
